infoextract.js

```diff
@@ -2,13 +2,53 @@ const axios = require("axios");
 const url = require("url");
 const querystring = require('querystring');
 const moment = require("moment");
+const storage = require("./storage");
 
 const YOUTUBE_API_URL = "https://www.googleapis.com/youtube/v3";
 const YtApi = axios.create({
 	baseURL: YOUTUBE_API_URL,
 });
 
 module.exports = {
+	/**
+	 * Gets all necessary information needed to represent a video. Handles
+	 * local caching and obtaining missing data from external sources.
+	 * Note that this function does not update the cache. That should be
+	 * handled by the functions that actually retrieve missing info.
+	 * @param	{string} service The service that hosts the source video.
+	 * @param	{string} id The id of the video on the given service.
+	 * @return	{Object} Video object
+	 */
+	getVideoInfo(service, id) {
+		// TODO: check if service is valid
+		// TODO: check if id is valid for service
+		return storage.getVideoInfo(service, id).then(result => {
+			let video = {};
+			for (const key in result) {
+				if (result.hasOwnProperty(key)) {
+					video[key] = result[key];
+				}
+			}
```

Comment:
What is the point of checking if `result` has a property `key`? Wouldn't this be better written as `const key in Object.keys(result)`?

Additionally, updating an object with partial properties sounds like a perfect use case for [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

Reply:
The `if (result.hasOwnProperty(key))` came from the vscode template. And yes, `Object.assign` appears to be the better solution.